x/market/keeper/grpc_query.go

```diff
@@ -0,0 +1,211 @@
+package keeper
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkquery "github.com/cosmos/cosmos-sdk/types/query"
+	"github.com/ovrclk/akash/x/market/types"
+)
+
+// Querier is used as Keeper will have duplicate methods if used directly, and gRPC names take precedence over keeper
+type Querier struct {
+	Keeper
+}
```

Comment:
Can you make `Querier` an interface, with a private struct implementation for all of these?

I'd like to do the same for `Keeper` as well.